application/helpers/remotecontrol/remotecontrol_handle.php

```diff
@@ -3115,107 +3115,105 @@ protected function _checkSessionKey($sSessionKey)
+            ->findAll();
 
-            // check if email is valid
-            $this->_checkEmailFormat($aData['email']);
+        foreach ($participants as $participant) {
 
-            if ($bIsValidEmail == true) {
+            $model = null;
 
-                //First, process the known fields
-                if (!isset($aData['participant_id']) || $aData['participant_id'] == "") {
-                    //  $arParticipantModel = new Participant();
-                    $aData['participant_id'] = Participant::gen_uuid();
-                }
-                if (isset($aData['emailstatus']) && trim($aData['emailstatus'] == '')) {
-                    unset($aData['emailstatus']);
-                }
-                if (!isset($aData['language']) || $aData['language'] == "") {
-                    $aData['language'] = "en";
-                }
-                if (!isset($aData['blacklisted']) || $aData['blacklisted'] == "") {
-                    $aData['blacklisted'] = "N";
-                }
-                $aData['owner_uid'] = Yii::app()->session['loginID'];
-                if (isset($aData['validfrom']) && trim($aData['validfrom'] == '')) {
-                    unset($aData['validfrom']);
-                }
-                if (isset($aData['validuntil']) && trim($aData['validuntil'] == '')) {
-                    unset($aData['validuntil']);
-                }
+            if (isset($participant['id'])) {
+                $participant['participant_id'] = $participant['id'];
+            }
+
+            if (isset($participant['participant_id'])) {
+                $model = Participant::model()->findByPk($participant['participant_id']);
+            } else {
+                $model = Participant::model()->findByAttributes(array(
+                    'firstname' => $participant['firstname'],
+                    'lastname'  => $participant['lastname'],
+                    'email'     => $participant['email'],
+                    'owner_uid' => Yii::app()->session['loginID']
+                ));
+            }
 
-                if (!empty($aData['email'])) {
-                    //The mandatory fields of email, firstname and lastname
-                    $sMandatory++;
+            // Participant not found, so we create a new one
+            if (!$model) {
+                $model = new Participant();
+                if (isset($participant['participant_id'])) {
+                    $model->participant_id = $participant['participant_id'];
+                } else {
+                    $model->participant_id = Participant::gen_uuid();
                 }
+            }
 
-                // Write to database if record not exists
-                if (empty($arRecordExists)) {
-                    // save participant to database
-                    Participant::model()->insertParticipantCSV($aData);
+            $scenario = $model->getScenario(); // insert or update
+            if ($scenario == 'update' && $update === false) {
```

Comment:
Strange to use getScenario :)

Why not , maybe more simple to understand 

````
if (!$model) {
    // Create
} elseif (!$update) {
    continue;
}
````

Reply:
Still think i prefer the elseif method. What it's your opinion @olleharstedt ?

Reply:
No strong opinion. So will merge.